pulsar/consumer_partition.go

```diff
@@ -479,7 +505,49 @@ func (pc *partitionConsumer) MessageReceived(response *pb.CommandMessage, header
 		return err
 	}
 
-	uncompressedHeadersAndPayload, err := pc.Decompress(msgMeta, headersAndPayload)
+	decryptedPayload, err := pc.decryptor.Decrypt(headersAndPayload.ReadableSlice(), pbMsgID, msgMeta)
+	messages := make([]*message, 0)
+
+	// error decrypting the payload
+	if err != nil {
+		pc.log.Error(err)
+		switch pc.decryptor.CryptoFailureAction() {
+		case crypto.ConsumerCryptoFailureActionFail:
+			pc.log.Errorf("consuming message failed due to decryption err :%v", err)
+			return err
+		case crypto.ConsumerCryptoFailureActionDiscard:
+			pc.discardCorruptedMessage(pbMsgID, pb.CommandAck_DecryptionError)
+			return fmt.Errorf("discarding message on decryption error :%v", err)
+		case crypto.ConsumerCryptoFailureActionConsume:
+			pc.log.Warnf("consuming encrypted message due to error in decryption :%v", err)
+			messages = append(messages, &message{
+				publishTime:  timeFromUnixTimestampMillis(msgMeta.GetPublishTime()),
```

Comment:
Nit. we can use create a new variable here and move `messages := make([]*message, 0)` back to where it was
```
messages := []*message{
		{
			
		},
	}
```

Reply:
done